src/test/java/com/iota/iri/storage/TangleTest.java

```diff
@@ -69,6 +70,19 @@ public void getKeysStartingWithValue() throws Exception {
         //Assert.assertNotEquals(tag.length, 0);
     }
 
+    @Test
+    public void saveTxsWithColludingHashes() throws Exception {
+        int[] trits = getRandomTransactionTrits();
+        Hash hash = Hash.calculate(SpongeFactory.Mode.CURLP81, trits);
+        TransactionViewModel goodTx = new TransactionViewModel(trits,hash);
+        goodTx.store(tangle);
+        TransactionViewModel colludingTx = new TransactionViewModel(getRandomTransactionTrits(), hash);
+        colludingTx.store(tangle);
```

Comment:
This is really testing TransactionViewModel:
```java
    public boolean store(Tangle tangle) throws Exception {
        if (hash.equals(Hash.NULL_HASH) || exists(tangle, hash)) {
            return false;
        }
```

Reply:
You are correct!

I opened #796 and I am closing this PR for now.